x-pack/plugins/security/server/routes/authentication/saml.ts

```diff
@@ -43,11 +50,15 @@ export function defineSAMLRoutes({ router, logger, authc, csp, basePath }: Route
       options: { authRequired: false },
     },
     (context, request, response) => {
+      // NodeJS limits the maximum size of the Request-Line + all HTTP headers to 8KB.
+      // See https://nodejs.org/api/cli.html#cli_max_http_header_size_size.
+      const maxRedirectURLSize = config.authc.saml?.maxRedirectURLSize.getValueInBytes() ?? 4096;
```

Comment:
__note:__ it's technically possible to call this endpoint even if SAML is not enabled (no use case for that yet), so that config isn't available. I thought not throwing error would be better, but I don't have strong opinion here.